staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -46,6 +46,20 @@ import (
 // TODO(lavalamp): I believe there is a possible race only when using an
 //                 external known object source that the above TODO would
 //                 fix.
+//                 Whether there is a potential race depends on how the comsumer
+//                 modifies knownObjects. In Pop(), process function is called under
+//                 lock, so it is safe to update data structures in it that need to be
+//                 in sync with the queue (e.g. knownObjects).
+//
+//                 Example:
+//                 In case of sharedIndexInformer being a consumer
+//                 (https://github.com/kubernetes/kubernetes/blob/0cdd940f/staging/
+//                 src/k8s.io/client-go/tools/cache/shared_informer.go#L192),
+//                 there is no race as knownObjects (s.indexer) is modified safely
+//                 under DeltaFIFO's lock. The only exceptions are GetStore() and
+//                 GetIndexer() methods, which expose ways to modify the underlying
+//                 storage. Currently these two methods are used for creating Lister
+//                 and internal tests.
```

Comment:
I spend 30 minutes looking into this code and this comment looks correct to me.
However, I would prefer someone else also to take a look.